pkg/kubelet/prober/worker.go

```diff
@@ -316,11 +316,14 @@ func (w *worker) doProbe(ctx context.Context) (keepGoing bool) {
 
 	w.resultsManager.Set(w.containerID, result, w.pod)
 
-	if (w.probeType == liveness || w.probeType == startup) && result == results.Failure {
+	if (w.probeType == liveness && result == results.Failure) || w.probeType == startup {
 		// The container fails a liveness/startup check, it will need to be restarted.
 		// Stop probing until we see a new container ID. This is to reduce the
 		// chance of hitting #21751, where running `docker exec` when a
 		// container is being stopped may lead to corrupted container state.
+		// In addition, if the container succeeds a startup probe, we should stop probing
```

Comment:
I don't see any status check for a startup probe here anymore. In fact this change removes result checking for startup probe if I understood it correctly. Can you elaborate on this a little bit more?

Reply:
@bart0sh 
Thank you for your comment!

The main point of this change is stopping `startupProbe` if it exceeds the threshold of each status(`Success` or `Failure`).
In the previous section, we will check the status and threshold, so if this condition is not met, it is assumed that the threshold is exceeded for each status.

```go
	if (result == results.Failure && w.resultRun < int(w.spec.FailureThreshold)) ||
		(result == results.Success && w.resultRun < int(w.spec.SuccessThreshold)) {
		// Success or failure is below threshold - leave the probe state unchanged.
		return true
	}
```
However, strictly speaking, there is a status called `Unknown`, so I've a check for explicitly `Success` or `Failure` and updated comment.